### ISSUE TITLE
Multiple bounds are confused

### ISSUE THREAD

**Issue by alecmocatta** [magiclen/educe#1]
Hey @magiclen, thanks for the handy crate!

Handling multiple (different) bounds seems to be broken. This:

```rust
#[derive(Educe)]
#[educe(Clone(bound = "T: Clone"), Default(bound = "T: Default"))]
struct Type<T>(T);
```

Expands to:
```rust
impl<T> core::default::Default for Type<T>
where
    T: Clone,
{
    #[inline]
    fn default() -> Self {
        Type(<T as core::default::Default>::default())
    }
}
impl<T> core::clone::Clone for Type<T>
where
    T: Default,
{
    #[inline]
    fn clone(&self) -> Self {
        Type(core::clone::Clone::clone(&self.0))
    }
    #[inline]
    fn clone_from(&mut self, _source: &Self) {
        core::clone::Clone::clone_from(&mut self.0, &_source.0);
    }
}
```

Which gives a bunch of errors:

```
error[E0277]: the trait bound `T: std::default::Default` is not satisfied
 --> src/main.rs:3:10
  |
3 | #[derive(Educe)]
  |          ^^^^^ the trait `std::default::Default` is not implemented for `T`
  |
  = note: required by `std::default::Default::default`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
  |
3 | #[derive(Educe + std::default::Default)]
  |                ^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `T: std::clone::Clone` is not satisfied
 --> src/main.rs:3:10
  |
3 | #[derive(Educe)]
  |          ^^^^^ the trait `std::clone::Clone` is not implemented for `T`
  |
  = note: required by `std::clone::Clone::clone`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
  |
3 | #[derive(Educe + std::clone::Clone)]
  |                ^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `T: std::clone::Clone` is not satisfied
 --> src/main.rs:3:10
  |
3 | #[derive(Educe)]
  |          ^^^^^ the trait `std::clone::Clone` is not implemented for `T`
  |
  = note: required by `std::clone::Clone::clone_from`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
  |
3 | #[derive(Educe + std::clone::Clone)]
  |                ^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors
```

**Comment by magiclen** [comment]
Thanks for the report. It was a really stupid bug which has been fixed now.

**Comment by alecmocatta** [comment]
Confirmed fixed by a88f71f6369d84a54f7caeeb23682820e906ec72, thanks!
